_posts/2018-05-15-jiractl.md

```diff
@@ -0,0 +1,105 @@
+---
+layout: post
+title: "jiractl: A command-line tool for managing Jira"
+date: 2018-05-15 08:53:01 -0800
+cover: /assets/images/jiractl-cover.jpg
+excerpt: This post introduces jiractl, a command-line tool for managing Jira. We provide some instructions on how to set up and use jiractl.
+authors:
+  - name: Emma Lubin
+    url: https://twitter.com/lubin_emma
+    photo: https://avatars1.githubusercontent.com/u/3219008?s=400&v=4
+  - name: Gabe Isenberg
+    url: https://twitter.com/the_gisenberg
+    photo: https://avatars0.githubusercontent.com/u/641505?s=400&v=4
+  - name: Jon Yonker
+    url: https://www.linkedin.com/in/jonyonker/
+    photo: https://avatars3.githubusercontent.com/u/393581?s=400&v=4
+  - name: Oliver Colbert
+    url: https://www.linkedin.com/in/oliverbcolbert/
+    photo: /assets/images/ocolbert.png
+  - name: Sam Hughes
+    url: https://www.linkedin.com/in/whatsupbucho/
+    photo: /assets/images/shughes.png
+---
+
+At GoDaddy we use [Jira](https://www.atlassian.com/software/jira) to track and manage our projects. Jira has an extensive feature set that can be customized to different teams’ needs, but its customizability can make it difficult for developers to perform simple tasks during a sprint. Visiting the Jira UI to check sprint progress or update an issue takes a developer out of their normal workflow on the command line. During a recent hackathon, we decided to address this problem by building an opensource Jira CLI for managing our sprints.
+
+There are a number of other Jira CLIs already out there; Atlassian sells one, there are a handful of opensource node-based Jira CLIs -- for example, [jira-node-cli](https://github.com/lusarz/jira-node-cli), [jira-console](https://github.com/faressoft/jira-console), and [JiraTool](https://github.com/WillBrock/JiraTool) -- and [go-jira](https://github.com/Netflix-Skunkworks/go-jira), a highly-starred tool written in Go.
+
+We found existing tools that met some of our needs, but decided to contribute our own to ensure we could support all of our common use cases and requirements:
+* Easy management of multiple teams
+* Intuitive, context-based syntax
+* Board, epic, sprint, and issue views
+* Issue creation from template files
+
+### Features
+
+_Support for multiple teams_ -- One engineering group at GoDaddy might have multiple scrum teams each with their own Jira boards. We wanted to make it easy to manage and switch between boards within a project. After installing jiractl you can add a context for the Jira instance you use and fetch teams configured within that instance. For example, for GoDaddy:
+
+```console
+$ npm i @godaddy/jiractl --global
```

Comment:
nit: It seems inconsistent to use the shorthand `i` for install but the verbose `--global` for global.